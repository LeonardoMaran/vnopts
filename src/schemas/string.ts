import { Schema, SchemaParameters } from '../schema';
import { ValidateResult } from '../types';

export class StringSchema extends Schema<string, SchemaParameters<string>> {
  public expected() {
    return 'a string';
  }
  public validate(value: unknown): ValidateResult {
    return typeof value === 'string';
  }
}
