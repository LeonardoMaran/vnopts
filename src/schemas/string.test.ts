import { eachHandler } from '../../tests/__helpers__/utils';
import { StringSchema } from './string';

const name = '<key>';
const value = 'string';

describe.each`
  parameters | input                | output               | hasWarnings
  ${{}}      | ${{ [name]: value }} | ${{ [name]: value }} | ${false}
  ${{}}      | ${{ [name]: 1 }}     | ${Error}             | ${false}
`('', eachHandler<StringSchema>(StringSchema, { name }));
