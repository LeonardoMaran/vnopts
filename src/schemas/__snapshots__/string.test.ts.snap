// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema {"name":"<key>"} with input {"<key>":1} 1`] = `"Invalid \\"<key>\\" value. Expected a string, but received 1."`;
